### ISSUE TITLE
Rename SimpleStore to StoreBase

### ISSUE THREAD

**Issue by CatoLeanTruetschel** [amosproj/amos-ss2021-project2-context-map#241]
Rename `SimpleStore` to `StoreBase` as there is no complex store and this seems to be nothing else but the common base type of all stored.

**Comment by joluj** [comment]
A 'complex' store would be a store where the `setState` accepts an different argument type than `getState` outputs (e.g. when he maps the input from `setState` to a new state or where the pure `setState` is not allowed; @PianoRollRepresentation creates one during the refactoring process of the Filter)

A `StoreBase` would be nice as well, but then as the interface

```ts
abstract class StoreBase<T> {
  protected readonly storeSubject = ...;
  protected abstract getInitialValue(): T;
  public getState(): Observable<T> {
    return this.storeSubject.pipe();
  }
}
```

TLDR: Naming is fine. One additional abstraction would be benefitial.
